tool/tctl/common/auth_command.go

```diff
@@ -748,6 +754,53 @@ func (a *AuthCommand) checkProxyAddr(clusterAPI auth.ClientI) error {
 	return trace.BadParameter("couldn't find registered public proxies, specify --proxy when using --format=%q", identityfile.FormatKubernetes)
 }
 
+func (a *AuthCommand) logUngrantedTTL(clusterAPI auth.ClientI) error {
+	rc := clockwork.NewRealClock()
+	var addr string
+	var u *utils.NetAddr
+	if a.proxyAddr != "" {
+		var err error
+		u, err = utils.ParseAddr(a.proxyAddr)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		addr = u.Addr
+	}
+	status, err := client.StatusFor("", addr, a.genUser)
+	if err != nil && !trace.IsNotFound(err) {
+		return trace.Wrap(err)
+	}
+	if status != nil && !status.IsExpired(rc) {
+		remaining := status.ValidUntil.Sub(rc.Now())
+		if a.genTTL > remaining {
+			logrus.Warnf("requested ttl of %s cannot be granted. "+
+				"Existing session has a TTL that ends in %s. proceeding with %s ttl.",
+				a.genTTL, remaining, remaining,
+			)
+		}
+		return nil
+	}
+	user, err := clusterAPI.GetUser(a.genUser, true)
```

Comment:
This is a lot of client-side logic for a few log messages, including things I assume we already check during cert generation (parsing proxy address, checking statuses, etc). Could we simplify it?

A few ideas:
* if we print the messages _after_ the certs are generated, could we drop some of the checks?
* if we wait for cert generation, could we check if the user got the requested TTLs (and hence skip all this)?
* could we get away with a simpler message and still address the issue (say, dropping the need to read roles)?

Something else that strikes me is that, if any of this fails, we are failing cert generation. That seems a bit off.

Reply:
> if we wait for cert generation, could we check if the user got the requested TTLs (and hence skip all this)?

This seems like the most sensible option to me, however (I'm fairly sure I'm missing something here)  when I get the valid time for the cert from `key.TeleportTLSCertValidBefore` its just using the requested ttl, despite my user having a max time ttl of 5 hours in all of their roles.

Calling tctl auth sign like this:
```
 tctl sign --format file --user admin --ttl 400h -o admin.identify
```
The ttl doesnt seem to be getting truncated to 5hours acording to the contents of the cert, it also doesnt get truncated after I `tsh login`.

Reply:
Huh, interesting. I assumed that would be the case, but apparently I'm wrong. :)

Reply:
I had actually set teleport up incorrectly I think so, that method works, I've updated it with an error message  that is a bit less specific but now doesnt require getting users/roles etc.